src/Functions/if.cpp

```diff
@@ -1,33 +1,34 @@
-#include <DataTypes/DataTypesNumber.h>
-#include <DataTypes/DataTypesDecimal.h>
+#include <type_traits>
```

Comment:
Personal preference, but I prefer keeping std headers in a different block and after project headers.

Reply:
It's the formatted result of clang-format. I'll seperate std headers in another group